iac-extensions/arm/src/main/java/org/sonar/iac/arm/checkdsl/ContextualResource.java

```diff
@@ -66,6 +71,27 @@ protected HasTextRange toHighlight() {
   }
 
   public ContextualResource childResourceBy(String type, Predicate<ResourceDeclaration> predicate) {
+    return Optional.ofNullable(nestedChildResourceBy(type, predicate))
+      .filter(ContextualResource::isPresent)
+      .orElseGet(() -> childResourceOutsideOfThisBy(type, predicate));
+  }
+
+  private ContextualResource childResourceOutsideOfThisBy(String type, Predicate<ResourceDeclaration> predicate) {
+    var topLevelResources = Optional.ofNullable(tree)
+      .map(ResourceDeclaration::parent)
+      .map(ContextualResource::getChildResources)
+      .orElse(Stream.empty());
+
+    return topLevelResources
+      .filter(this::isExternalChildOfThis)
+      .filter(it -> TextUtils.isValue(it.type(), this.type + "/" + type).isTrue())
+      .filter(predicate)
+      .findFirst()
```

Comment:
This call occurs already after all of the filtering: getting external children, checking type and applying the predicate is applied on the stream, so this will return only what was actually requested by caller.